src/app/child-dev-project/previous-schools/previous-schools.component.spec.ts

```diff
@@ -76,4 +83,40 @@ describe("PreviousSchoolsComponent", () => {
       done();
     });
   });
+
+  it("should only show columns which are defined by the config", fakeAsync(() => {
+    const config: PanelConfig = {
+      entity: new Child(),
+      config: {
+        single: true,
+        columns: {
+          schoolId: "Team",
```

Comment:
It's not intuitively clear that these are assigning a label to the entity property key. An array of column objects (`[{ id: "schoolId", label: "Team" }]`) may be more explicit and extendable - although a bit more verbose as well.

In general there is a question here, as Andy also raised it, whether generalization (and its hard to understand ways of pushing around config) trumps some duplication with explicit, obvious implementations for teams and schools

Reply:
I this case I chose a little different approach for the config, because the options are quite limited and only the mentioned five are available. Theoretically, we could also make the `ChildSchoolRelation` a configurable entity and allow to display any number of columns in this view.

Regarding your second point: As long as we are using the exact same classes to model teams and schools, as well as the relations between the children and their teams/schools, I do not see a reason why we should use different components to display these relations. The visualization differences are only additional fields with no further special meaning. The only real difference, which we now also had in the case were classes were modeled through `School` objects, is whether a child can have many active relations or not. Therefore, I would not expect to have different components for teams and schools but rather single and multiple active relations which in this case is configured using the `single` flag.